packages/react/src/hero/hero.tsx

```diff
@@ -1,33 +1,139 @@
-import { cx } from 'cva';
+import { type VariantProps, cva, cx } from 'cva';
 import type { HTMLProps } from 'react';
+import { GroupContext, Provider } from 'react-aria-components';
 
-type HeroProps = HTMLProps<HTMLDivElement> & {
+type Variants = VariantProps<typeof variants>;
+
+type ExcludeVariants<T> = {
+  [K in keyof Variants]?: Exclude<Variants[K], T | undefined>;
+};
+
+type HeroBaseProps = HTMLProps<HTMLDivElement> & {
   children: React.ReactNode;
 };
 
-const Hero = ({ className, children }: HeroProps) => {
-  return (
-    <div
-      className={cx(
-        // Wrap text content in a container
-        '*:data-[slot="content"]:container',
-        // Style the Hero heading
-        '**:data-[slot="heading"]:heading-l',
-        // Spacing between the media and the text content above it
-        '*:data-[slot="media"]:mt-8 lg:*:data-[slot="media"]:mt-10',
-        // Responsive heights for the <Media> wrapper component
+type HeroProps1 = HeroBaseProps & ExcludeVariants<'full-bleed' | 2>;
+
+type HeroProps2 = HeroBaseProps & ExcludeVariants<'two-column' | 1>;
+
+type HeroProps = HeroProps1 | HeroProps2;
```

Comment:
er det kanskje greit med litt bedre navngivning på HeroProps1 og HeroProps2? 😅 vet det er interne props men greit for neste person kanskje. Ikke så viktig men tenkte si ifra i det minste

Reply:
haha ja, litt mer verbose gir mer mening, men den er grei :D